### PR TITLE
[WIP] [DMN] Missions

### PR DESCRIPTION
**Pull request by alexkar598**
- [ ] Deliver item (pick up a special object, fly to a station and delivery it by clicking a console with it)

- [ ] Defend colony (wave attack, waves get harder as you go, score awarded by how damaged the colony is)

- [ ] Board ship to download tactical data 

- [ ] Combat thing with a freighter escorted by a bunch of enemies. You have to beam over to the freighter and find the bio agent which you must then destroy

- [ ] A player is spawned in on a small, weak ship as a high ranking romulan military admiral. He's given the objective "you have betrayed your romulan allies, survive at all costs". Their ship is just a standard AI ship with an interior, but the admiral has no control over it. Admiral gets a holopad to contact the station. When the ship gets close, an announcement plays explaining that they're an admiral who wants to defect the pilot gets an option to either help the admiral, or leave him. If help is selected his shields drop and you can bring him aboard but enemies spawn
If you leave them you fail the mission

:cl: optional name here
add: Space UPS,new mission: item delivery
/:cl:

WIP not done,item delivery is almost done.

### CODE REVIEW

**Comment by soryy708** [comment]
This is going to become one mega PR. Each of these checkboxes can be its own PR if atomized.

File: DS13/code/modules/overmap/events.dm [hunk 1]
```diff
@@ -351,4 +351,107 @@ GLOBAL_LIST_INIT(overmap_event_spawns, list())
...
+	icon_state = "sealedcrate"
+	var/datum/overmap_event/linked_event
+	max_integrity = 120
+	density = TRUE
+
+/obj/structure/sealedcrate/attack_hand(mob/user)
+	. = ..()
+	if(.) 
+		return
+	to_chat(user,"<span class='warning'>This crate is locked and sealed,you can't open it.</span>") //HEY LEAVE THIS ALONE OR YOUR SCRUBBING PLASMA CONDUITS
```
Comment (soryy708) [inline DS13/code/modules/overmap/events.dm:390]: Missing space after `,`

File: DS13/code/modules/overmap/events.dm [hunk 2]
```diff
@@ -351,4 +351,107 @@ GLOBAL_LIST_INIT(overmap_event_spawns, list())
...
+		SC.linked_event.succeed()
+		while(SC.alpha != 0)
+			SC.alpha = min(SC.alpha-1,0)
+		qdel(SC)
+
+//DATUM
+
+/datum/overmap_event/deliver_item
+	name = "Item delivery"
+	desc = "YEEEEEET,CALL A CODER AT 1800-CODERRRRRRRRRRRRR.var/desc not set,check new() proc for runtimes."
```
Comment (soryy708) [inline DS13/code/modules/overmap/events.dm:420]: Autistic.

**Comment by alexkar598** [comment]
I might rework the event system in this PR which will make atomizing this shit hard.

**Comment by alexkar598** [comment]
Not going to do all this in this pr,il make another one when I'm home and the code is pushed.

**Pull request closed without merging**